docs/integrations/webhooks/debug-your-webhooks.mdx

```diff
@@ -0,0 +1,112 @@
+---
+title: Debug your webhooks
+description: Understand how to debug your webhooks while developing your application
+---
+
+# Debug your webhooks
+
+Developing with webhooks can be a new experience for developers. It can be hard to debug when something goes awry. This guide will cover the basics for debugging and help you to direct your attention to the correct spot.
+
+## Webhooks and local development
+
+When you or a user of your application performs certain actions, a webhook can be triggered. You can see the full list of [webhook events](/docs/integrations/webhooks/overview#supported-webhook-events) for a list of the actions that could result in a Webhook. Depending on the events subscribed to in the [Webhook section](https://dashboard.clerk.com/last-active?path=webhooks) of your dashboard, a webhook event will be triggered and sent to the specified endpoint in your application.
+
+When you are developing on your localhost, your application is not internet facing and can't receive the webhook request. You will need to use a tool that creates a tunnel from the internet to your localhost. These tools will provide temporary or permanant URLs depending on the tool and the plan you subscribe to. Popular tools include `ngrok`, `localtunnel` and `Cloudflare Tunnel`.
+
+Below is an image demonstrating the flow from when you perform an action that triggers a webhook through until the webhook is received by your application. This flow chart will show the role of the tunnel.
+
+FLOW CHART IMAGE HERE
+
+With the Webhook flow in mind, the following areas could cause problems when trying to develop with webhooks:
+
+1. Middleware is not configured correctly for webhooks
```

Comment:
Link these options to their corresponding sections, so `[Middleware is not configured correctly for webhooks](#check-your-middleware-configuration)`